modules/samples/extensions/geo-event-fuser/src/main/java/org/wso2/siddhi/extension/geo/geoeventfuser/FuseEvents.java

```diff
@@ -100,46 +99,38 @@ protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processo
         nextProcessor.process(streamEventChunk);
     }
 
+    /**
+     * Get number of deployed execution plans
+     *
+     * @return number of deployed execution plans
+     */
     public Integer getDeployedExecutionPlansCount() {
         return ExecutionPlansCount.getNumberOfExecutionPlans();
     }
 
+    /**
+     * Fuse given {@link StreamEvent}.
+     * -Precedence of states when fusing-
+     * goes low to high from LHS to RHS
+     * OFFLINE < NORMAL < WARNING < ALERT
+     * States will be in all caps to mimics
+     * that states not get on the way
+     *
+     * @param event {@link StreamEvent} to fuse
+     */
     public void fuseEvent(StreamEvent event) {
-
-    /*
-        * --For reference--
-        * -Precedence of states-
-        * goes low to high from LHS to RHS
-        *
-        * OFFLINE < NORMAL < WARNING < ALERT
-        * */
-
-
-    /*
-        * --For reference--
-        *   Higher the index higher the Precedence
-        *   States are in all caps to mimics that states not get on the way
-        *
-        * */
-
-        String[] statesArray = new String[] { "OFFLINE", "NORMAL", "WARNING", "ALERTED" };
+        String eventId = (String) event.getOutputData()[eventIdPosition];
+        ArrayList<StreamEvent> receivedEvents = eventsBuffer.get(eventId);
         List<String> states = Arrays.asList(statesArray);
-
-        Object[] data = event.getOutputData();
-
+        Object[] outputData = event.getOutputData();
         String finalState = "";
         String information = "";
-
-        String eventId = (String) event.getOutputData()[variablePosition];
-        ArrayList<StreamEvent> receivedEvents = eventsBuffer.get(eventId);
-
         String alertStrings = "";
         String warningStrings = "";
 
         Integer currentStateIndex = -1;
-
         for (StreamEvent receivedEvent : receivedEvents) {
-            String thisState = (String) receivedEvent.getOutputData()[8];
+            String thisState = (String) receivedEvent.getOutputData()[statePosition];
             Integer thisStateIndex = states.indexOf(thisState);
```

Comment:
@mohanvive `thisState` meant the state of a received event. In here, `currentState` would mean the final state of the fused event. Therefore, in #195 `thisState` was refactored to `eventState` to get rid of the confusion.